aten/src/ATen/native/cuda/UnaryGeometricSinKernel.cu

```diff
@@ -22,33 +22,43 @@ void sin_kernel_cuda(TensorIteratorBase& iter) {
 #if AT_USE_JITERATOR()
     static const auto sin_string = jiterator_stringify(
         template <typename T> T sin_impl(T a) { return std::sin(a); });
-    AT_DISPATCH_COMPLEX_TYPES_AND(
-        kComplexHalf, common_dtype, "sin_name", [&]() {
+    AT_DISPATCH_V2(
+        common_dtype,
+        "sin_name",
+        AT_WRAP([&] {
           jitted_gpu_kernel<
               /*name=*/sin_name,
               /*return_dtype=*/scalar_t,
               /*common_dtype=*/scalar_t,
               /*arity=*/1>(iter, sin_string);
-        });
+        }),
+        kComplexHalf,
```

Comment:
you forgot AT_EXPAND(AT_COMPLEX_TYPES) here

Reply:
My bad 😅